sc/shared/src/test/scala/sigma/LanguageSpecificationV5.scala

```diff
@@ -7957,77 +7957,81 @@ class LanguageSpecificationV5 extends LanguageSpecificationBase { suite =>
           )
         )
     )
-    verifyCases(
-      // (coll, (index, default))
-      {
-        def success[T](v: T) = Expected(Success(v), 1773, costDetails, 1773)
-        Seq(
-          ((Coll[Int](), (0, default)), success(default)),
-          ((Coll[Int](), (-1, default)), success(default)),
-          ((Coll[Int](1), (0, default)), success(1)),
-          ((Coll[Int](1), (1, default)), success(default)),
-          ((Coll[Int](1), (-1, default)), success(default)),
-          ((Coll[Int](1, 2), (0, default)), success(1)),
-          ((Coll[Int](1, 2), (1, default)), success(2)),
-          ((Coll[Int](1, 2), (2, default)), success(default)),
-          ((Coll[Int](1, 2), (-1, default)), success(default))
-        )
-      },
-      existingFeature((x: (Coll[Int], (Int, Int))) => x._1.getOrElse(x._2._1, x._2._2),
-        "{ (x: (Coll[Int], (Int, Int))) => x._1.getOrElse(x._2._1, x._2._2) }",
-        if (lowerMethodCallsInTests)
-          FuncValue(
-            Vector((1, SPair(SCollectionType(SInt), SPair(SInt, SInt)))),
-            BlockValue(
-              Vector(
-                ValDef(
-                  3,
-                  List(),
-                  SelectField.typed[Value[STuple]](
-                    ValUse(1, SPair(SCollectionType(SInt), SPair(SInt, SInt))),
-                    2.toByte
+
+    if(!VersionContext.current.isV6SoftForkActivated) {
```

Comment:
Protocol change like this can be tested using changeFeature (see other tests) in combination with newVersionedResults (also see code). In this case both old and new behaviour will be tested.

In principle this test case can be moved to LSV6.

Reply:
changedFeature is not enough here, you need for versioned verifyCases, as in 6.0 costing trace would be different in some cases due to difference in default evaluation (at the same time, it wouldnt provide higher cost, so the change is secure costing-wise). 

Seems like *Feature machinery is overly complex, but still not enough to describe protocol changes, thus I would recommend to decide what it is actually testing (or should test), and then follow simpler approaches

Reply:
Please see other examples where changedFeature is used, there are test cases where cost trace depend on the version and you can pass different traces to the constructor of Expected.
The perceived complexity is due to learning curve.

Reply:
changedFeature is tied to V4/V5 switch, see checkExpected / checkVerify functions in SigmaDslTesting. So it cant be used in V6 testing without modifications it seems. So it is definitely not about learning curve, changedFeature is just not ready for V6

Reply:
@aslesarenko see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/1024 for example

Reply:
done